src/commands/utils/schemas.ts

```diff
@@ -0,0 +1,138 @@
+ * @param schema The Schema/version to delete.
+ * @param loader ResourceLoader to use to get the schema group.
+ * @returns Customized prompt for the delete action.
+ */
+export async function getDeleteSchemaVersionPrompt(
+  hardDeletion: boolean,
+  schema: Schema,
+  schemaGroup: Schema[],
+): Promise<string> {
+  const isOnlyVersion = schemaGroup.length === 1;
+
+  const deleteVerb = (hardDeletion ? "hard" : "soft") + " delete";
+  let prompt: string;
+
+  if (isOnlyVersion) {
+    prompt = `Are you sure you want to ${deleteVerb} the only version of subject ${schema.subject}?`;
+  } else {
+    const isLatestVersion = schema.version === schemaGroup[0].version;
+    if (isLatestVersion) {
+      prompt = `Are you sure you want to ${deleteVerb} the latest version of subject ${schema.subject}? Version ${schemaGroup[1].version} will become the latest.`;
+    } else {
+      prompt = `Are you sure you want to ${deleteVerb} version ${schema.version} of subject ${schema.subject}?`;
+    }
+  }
+
+  return prompt;
+}
+
+/** Function signature for vscode input validators */
+export type ValidationFunction = (input: string) => vscode.InputBoxValidationMessage | undefined;
+
+/**
+ * Returns a pair:
+ *  [0]: Function that can be used to validate the confirmation input for a schema version for deletion.
+ *  [1]: The string to show in the input box as a placeholder.
+ * @param version The version to confirm.
+ * @param hardDeletion Whether this is for a hard deletion or not.
+ * @returns pair of [validation function, prompt string]
+ */
+export function getSchemaDeletionValidatorAndPlaceholder(
+  version: number,
+  hardDeletion: boolean,
+): [ValidationFunction, string] {
+  let validator: ValidationFunction;
+  let prompt: string;
+
+  if (hardDeletion) {
+    prompt = `Enter "hard v${version}" to confirm, escape to cancel.`;
+    validator = (input: string) => {
+      if (input === `hard v${version}`) {
+        return;
+      }
+      return {
+        message: `Enter "hard v${version}" to confirm hard deletion, escape to cancel.`,
+        severity: vscode.InputBoxValidationSeverity.Error,
+      };
+    };
+  } else {
+    prompt = `Enter "v${version}" to confirm, escape to cancel.`;
+    validator = (input: string) => {
+      if (input === `v${version}`) {
+        return;
+      }
+      return {
+        message: `Enter "v${version}" to confirm, escape to cancel.`,
+        severity: vscode.InputBoxValidationSeverity.Error,
+      };
+    };
+  }
+
+  return [validator, prompt];
+}
+
+/**
+ * Ask user if they wany hard or soft delete behavior.
+ *
+ * @returns {Promise<boolean>} - true if hard delete, false if soft delete; undefined if cancelled.
+ */
+export async function hardDeletionQuickPick(): Promise<boolean | undefined> {
+  const strengthStr = await vscode.window.showQuickPick(
+    [
+      "Soft Delete -- existing records will remain deserializable",
+      "Hard Delete -- any existing records will NOT be deserializable",
+    ],
```

Comment:
Minor UI improvement:
```suggestion
    [
      { label: "Soft Delete", description: "existing records will remain deserializable" },
      { label: "Hard Delete", description: "any existing records will NOT be deserializable" },
    ],
```

Reply:
oh nice, yay.

Reply:
But would then either require changing the return to `strengthStr.label.startsWith("Hard")` or you could use this to pass in a `value` to check instead of having to string-compare:
https://github.com/confluentinc/vscode/blob/7aeb2354158cfb2781263d46853fdc8f664f975c/src/quickpicks/types.ts#L8-L10